event_scheduler/event_scheduler.py

```diff
@@ -115,12 +138,15 @@ def start(self):
     def stop(self):
         with self._scheduler_status_lock:
             if self._scheduler_status != self.SchedulerStatus.RUNNING:
-                # TODO: Add error message
                 return -1
             self._scheduler_status = self.SchedulerStatus.STOPPING
-            super().enterabs(sys.maxsize, sys.maxsize, None)
-            # TODO: Fix this hackiness
-
+        super().enterabs(sys.maxsize, sys.maxsize, None)
+        # Notify the event thread about a new event in the queue (in this case,
+        # it's the thread terminating event)
+        with self._cv:
```

Comment:
can the method you defined above `_notify` be used here instead?

Reply:
Yeah we could my notify method instead.